src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -186,15 +186,23 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
             _coreShell.AssertIsOnMainThread();
             IsLoading = true;
 
-            var libPath = await GetLibPath();
-            await _packageManager.UninstallPackageAsync(package.Name, libPath);
-            await ReloadInstalledAndLoadedPackagesAsync();
+            if(package.IsLoaded) {
+                await UnloadAsync(package);
+            }
 
-            IsLoading = false;
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                ReplaceItems(_installedPackages);
-            } else if(_selectedTab == SelectedTab.LoadedPackages) {
-                ReplaceItems(_loadedPackages);
+            if (!package.IsLoaded) {
+                var libPath = await GetLibPath();
+                await _packageManager.UninstallPackageAsync(package.Name, libPath);
```

Comment:
I think here it also should be `await _packageManager.UninstallPackageAsync(package.Name, package.LibraryPath.ToRPath())`